### PR TITLE
WIP general cleanup

### PR DESCRIPTION
**Pull request by endophage**
Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

**Comment by diogomonica** [comment]
@endophage rebase please

File: client/helpers.go [hunk 1]
```diff
@@ -62,3 +63,36 @@ func nearExpiry(r *data.SignedRoot) bool {
 	plus6mo := time.Now().AddDate(0, 6, 0)
 	return r.Signed.Expires.Before(plus6mo)
 }
+
+func initRoles(kdb *keys.KeyDB, rootKey, targetsKey, snapshotKey, timestampKey data.PublicKey) error {
+	rootRole, err := data.NewRole("root", 1, []string{rootKey.ID()}, nil, nil)
```
Comment (diogomonica) [inline client/helpers.go]: I'm always confused if the minVersion is 0 or 1. Also we still need to figure that out in VerifyRoot.

Comment (endophage) [inline client/helpers.go]: this isn't a version, it's a signing threshold for the role

**Comment by endophage** [comment]
@diogomonica rebased

**Comment by NathanMcCauley** [comment]
LGTM if tests pass

**Pull request merged**